neuralpp/util/callable_util.py

```diff
@@ -188,7 +191,8 @@ def infer_sympy_object_type(
                 if (
                     len(sympy_object.args) == 0
                 ):  # len(sympy_object.args) could raise (e.g, len(sympy.Add.args))
-                    raise TypeError(f"expect function application {sympy_object}")
+                    type_dict[sympy_object] = float
```

Comment:
This is a hack and a step towards switching to a thinner typing system for SymPy (as mentioned in my doc). Basically now we try to not have a dictionary with every SymPyExpression. So this is dealing with the case when a variable is not in the dictionary and we default it to "float" (which is not quite correct and hacky, since cleaning up typing system for SymPy is a TODO and this PR is focused on the Poly feature).